benchmark_custom.py

```
@@ -0,0 +1,49 @@
+        # Generate proof for a chunk (e.g. chunk 10 if there are enough, otherwise chunk 0)
+        chunk_index = 10 if size_mb > 10 else 0
+        proof = generate_merkle_proof(file_path, chunk_index=chunk_index, chunk_size=chunk_size)
+        end_time = time.perf_counter()
+
+        proof_time = end_time - start_time
+        print(f"generate_merkle_proof ({size_mb:.2f} MB file, chunk {chunk_index}): {proof_time:.4f} seconds")
+
+        print("--- Benchmark Complete ---")
+        return root_time, proof_time
+
+    except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)